src/charm.py

```diff
@@ -101,34 +103,13 @@ def mongodb_config(self) -> MongoDBConfiguration:
         hosts = [self.get_hostname_by_unit(self.unit.name)] + [
             self.get_hostname_by_unit(unit.name) for unit in peers.units
         ]
-        external_ca, _ = self.tls.get_tls_files("unit")
-        internal_ca, _ = self.tls.get_tls_files("app")
-
-        return MongoDBConfiguration(
-            replset=self.app.name,
-            database="admin",
-            username="operator",
-            password=self.get_secret("app", "operator_password"),
-            hosts=set(hosts),
-            roles={"default"},
-            tls_external=external_ca is not None,
-            tls_internal=internal_ca is not None,
-        )
+        return self._get_mongodb_config_for_user(OperatorUser, set(hosts))
```

Comment:
It seems that you always cast to `Set`, consider changing the method's signature with `List` and cast to `Set` from within the said method